src/custom/components/AccountDetails/AccountDetailsMod.tsx

```diff
@@ -0,0 +1,417 @@
+import React /* , { useCallback, useContext } */ from 'react'
```

Comment:
@W3stside if it's for re-export, maybe we don't even need the MOD file?

We can define our rules somewhere?
One early draft would be:


1. We don't modify anything in `src`, with a few exceptions:

2. If it's for swapping one component for another that is used in a `src` file, we just change the path from relative to absolute, and add the custom component in `custom` dir.  This sounds more complicated than it is, one example is 

```js
// src/components/MyTable.js
 import { MyRow } from './MyRow'
//   ... code 
// ------------------------------------------------ 
```

For creating a custom `MyRow`
```js
// src/components/MyTable.js
//   Changed path to absolute, so our MyRow component is used instead
 import { MyRow } from './MyRow'
//   ... leave all the remaining code of MyTable intact
// ------------------------------------------------

// custom/components/MyRow.js
//   ... custom code
// ------------------------------------------------ 
```

3. **Extend or delegate**: If we want to modify a component slightly, chances are that this can be done just by:
* Declaring a new analog custom component. i.e `src/components/MyTable` --> `custom/components/MyTable`
* The custom component would use inheritance or delegation for overriding it behaviour or look and feel

4. **Mod file**: If we need to modify a component that has lots of code, and we just want to change a few things, and `3.` is not an option:
* We copy the whole file to our `custom` dir. i.e. `src/components/MyTable` ---> `src/components/MyTable/MyTableMod`
* We create a new component that extends the component we want to modify
* We add the minimal props that would get the job done. We TRY THE BEST WE CAN to:
a) modify only the minimum!
b) maintain the rule that the original file and the Mod version are still comparable 1-1 line by line. Sometimes we will remove code by commenting lines, but we'll leave the original line there.
c) if we need to add something, we'll try to do it minimally. A flag that conditionally render something, or a component that gets injected for example. Some example is `components/Menu/index.tsx`

Reply:
@anxolin yeah but it wasn't just re-export. I also moved some functions out of the body for performance reasons see here for an easier diff view:
https://www.diffchecker.com/851AOAUd

Reply:
though yes maybe most of this could maybe have been simplified

Reply:
mm, then do you think we can use the above approaches. I'm worried about the copy paste maintainability